### PR TITLE
UHF-9708 & UHF-10171: Survey styling and javascript functionalities

### PR DESCRIPTION
**Pull request by hyrsky**
# [UHF-9708](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9708) and [UHF-10171](https://helsinkisolutionoffice.atlassian.net/browse/UHF-10171)

## What was done

* Styling for the survey component.
* Javascript functionalities for the survey component.

## How to install

* Check installation instructions from the [helfi_platform_config PR](https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/773)

## How to test

* [ ] After successfully installing the feature and adding your first survey it should appear where you have configured it to appear. There is currently 2 second delay configured in the code to make testing easier. This delay needs to be adjusted to the correct one later. Make sure the survey pops up after the timer is done.
* [ ] The popup is configured so that it doesn't come up again after you have closed it so to get it back go to empty local-storages `helfi-settings` row.
* [ ] The survey should look similar to what in these designs: https://www.figma.com/file/KzpwV4jzKnAgk7gy6x8Yfz/Kysely-pop-up?type=design&node-id=2501-94&mode=design
* [ ] Make sure the translations work (remember to run _make drush-locale-update drush-cr_ if it hasn't been mentioned elsewhere)
* [ ] When the popup comes up make sure that the focus is moved to the survey title. If you close the popup the focus should return to where it was before the popup opened.
* [ ] You should be able to close the popup with Esc button.
* [ ] You shouldn't be able to scroll or move focus to the content on the back while the popup is in place.
* [ ] When the popup appears the background shouldn't move about 15px because the scrollbar disappears but should remain exactly in the same place. The scrollbar is offset with padding on body.
* [ ] Check that code follows our standards

## Continuous documentation

* [ ] This feature has been documented/the documentation has been updated
* [ ] This change doesn't require updates to the documentation

## Translations

* [x] Translations have been added to .po -files and included in this PR

## Other PRs

* https://github.com/City-of-Helsinki/drupal-hdbt/pull/1009
* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/773
* https://github.com/City-of-Helsinki/drupal-helfi-etusivu/pull/642

[UHF-9708]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9708?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[UHF-10171]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-10171?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by juho-lehmonen — APPROVED** [review]

**Comment by teroelonen** [comment]
This should be reviewed again since it now has UHF-10171 tasks changes in it.

**Review by juho-lehmonen — APPROVED** [review]

File: src/js/closable-surveys.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+      function showSurvey() {
+        const { uuid } = survey.dataset;
+        const hiddenSurveys = surveysToHide?.[surveyKey] || [];
+        const shouldShowSurvey = !uuid || !hiddenSurveys.includes(uuid);
+
+        if (shouldShowSurvey) {
+          survey.style.display = 'flex';
+          toggleNoScroll(true);
+          setBodyPaddingRight(true);
```
Comment (juho-lehmonen) [inline src/js/closable-surveys.js:59]: This should be moved above "toggleNoScroll(true)", otherwise some browsers return "0px" for the padding value.

**Review by juho-lehmonen — CHANGES_REQUESTED** [review]

File: src/js/closable-surveys.js (hunk 1 above)
Comment (teroelonen) [reply src/js/closable-surveys.js:58]: This should be fixed now 👍 Nicely noticed 💪

File: src/js/closable-surveys.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        if (uuid) {
+          storageManager.addValue(surveyKey, uuid);
+        }
+      }
+
+      function addFocusTrap() {
+        surveyFocusTrap = window.focusTrap.createFocusTrap('#helfi-survey', {
+          initialFocus: () => '#helfi-survey__title'
+        });
+        surveyFocusTrap.activate();
```
Comment (hyrsky) [inline src/js/closable-surveys.js:35]: Focus trap (or something else here?) seems to disable cookie consent banner buttons while survey popup is drawn under the banner.
image

**Review by teroelonen — APPROVED** [review]
Very nice 🦖

**Pull request merged**